plugin/src/com/microsoft/alm/plugin/external/visualstudio/VisualStudioTfvcClient.java

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root.
+
+package com.microsoft.alm.plugin.external.visualstudio;
+
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.util.SystemInfo;
+import com.intellij.openapi.util.text.StringUtil;
+import com.microsoft.alm.plugin.external.exceptions.VisualStudioClientVersionException;
+import com.microsoft.alm.plugin.external.models.ToolVersion;
+import com.microsoft.alm.plugin.services.PropertyService;
+import org.apache.commons.lang.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+public class VisualStudioTfvcClient {
+
+    private static final Logger ourLogger = Logger.getInstance(VisualStudioTfvcClient.class);
+
+    private static final String[] SUPPORTED_VS_VERSIONS = new String[] { "2019", "2017" };
+    private static final String[] SUPPORTED_VS_EDITIONS = new String[] { "Enterprise", "Professional", "Community" };
+    private static final String EDITION_RELATIVE_TF_EXE_PATH =
```

Comment:
It looks like we have a different paths for a different editions.

Reply:
They are relative to an "edition root" directory. E.g. `"C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\" + EDITION_RELATIVE_TF_EXE_PATH`

Do you think I could use better constant name here?

Reply:
TF_EXE_PATH_RELATIVE_TO_EDITION_ROOT